core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamService.java

```diff
@@ -93,7 +95,32 @@
 
   @PreAuthorize("hasPermission(#stream, 'DELETE')")
   public void delete(Stream stream) {
-    throw new UnsupportedOperationException();
+    val existing = unsecuredGet(stream.getKey());
+    if (!existing.isPresent()) {
+      throw new ValidationException("Can't delete " + stream.getKey() + " because it doesn't exist");
+    }
+    handlerService.handleDelete(stream);
+    StreamKey streamKey = stream.getKey();
+    StreamBinding streamBinding = new StreamBinding(
+            new StreamBindingKey(
+                    streamKey.getDomain(),
+                    streamKey.getName(),
+                    streamKey.getVersion(),
+                    null,
+                    null),
+            null, null);
+    streamBindingService.delete(streamBinding);
+    streamRepository.delete(stream);
```

Comment:
Question... 

I'm more than likely being stupid here but I still don't see where `Consumer` or `Producer` entities are being deleted. 

`streamBindingService.delete(streamBinding)` triggers the cascading delete of `[Consumer|Producer]Bindings`. But I'm not seeing a similar cascade from `Stream` to `Consumer`/`Producer`. 

In my mind this is needed, but there's a good chance I'm missing something.

Reply:
added delete for producer and consumer when deleting stream